src/compas/geometry/_primitives/cylinder.py

```diff
@@ -0,0 +1,293 @@
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import division
+
+from math import pi
+
+from compas.geometry import Plane
+
+__all__ = ['Cylinder']
+
+
+class Cylinder(object):
+    """A cylinder is defined by a plane, a radius and a height.
+
+    Attributes
+    ----------
+    plane: :class:`compas.geometry.Plane`
+        The plane of the cylinder.
+    radius: float
+        The radius of the cylinder.
+    height: float
+        The height of the cylinder.
+
+    Examples
+    --------
+    >>> from compas.geometry import Plane
+    >>> from compas.geometry import Cylinder
+    >>> plane = Plane([0, 0, 0], [0, 0, 1])
+    >>> cylinder = Cylinder(plane, 5, 7)
+
+    """
+
+    __slots__ = ['_plane', '_radius', '_height']
+
+    def __init__(self, plane, radius, height):
+        self._plane = None
+        self._radius = None
+        self._height = None
+        self.plane = plane
+        self.radius = radius
+        self.height = height
+
+    @classmethod
+    def from_data(cls, data):
+        """Construct a cylinder from its data representation.
+
+        Parameters
+        ----------
+        data : :obj:`dict`
+            The data dictionary.
+
+        Returns
+        -------
+        Cylinder
+            The constructed cylinder.
+
+        Examples
+        --------
+        >>> from compas.geometry import Cylinder
+        >>> from compas.geometry import Plane
+        >>> data = {'plane': Plane.worldXY().data, 'radius': 5., 'height': 7.}
+        >>> cylinder = Cylinder.from_data(data)
+
+        """
+        cylinder = cls(Plane.worldXY(), 1, 1)
+        cylinder.data = data
+        return cylinder
+
+    # ==========================================================================
+    # descriptors
+    # ==========================================================================
+
+    @property
+    def plane(self):
+        """Plane: The plane of the cylinder."""
+        return self._plane
```

Comment:
With the circle as attr, you can still use setter and getter for plane and radius
```suggestion
        return self.circle.plane
```